src/SettingsPanel.svelte

```diff
@@ -46,30 +47,57 @@
         {/if}
         {#each $formatsStore as format, formatIdx (format.id)}
             <tr out:fade>
-                <td
-                    ><input
-                        type="text"
-                        class="form-control"
-                        bind:value={format.name}
-                        on:input={saveFormats}
-                    /></td
-                >
-                <td>
-                    <input
-                        type="text"
-                        class="form-control"
-                        bind:value={format.rule}
-                        on:input={saveFormats}
-                    />
-                </td>
-                <td>
-                    <input
-                        type="text"
-                        class="form-control"
-                        bind:value={format.mime}
-                        on:input={saveFormats}
-                    />
-                </td>
+                {#if !supportsClipboard()}
+                    <td
+                        ><input readonly="readonly"
+                            type="text"
+                            class="form-control"
+                            bind:value={format.name}
+                            on:input={saveFormats}
+                        /></td
+                    >
+                    <td>
+                        <input readonly="readonly"
+                            type="text"
+                            class="form-control"
+                            bind:value={format.rule}
+                            on:input={saveFormats}
+                        />
+                    </td>
+                    <td>
+                        <input readonly="readonly"
```

Comment:
On firefox we should give the user some indicator why the mime type is readonly. A simple way to do this is to add a title attribute to the `<input>`: `title="MIME type is not supported in Firefox."`. This is what it would look like when the user hovers over the readonly input:

<img width="894" alt="Screen Shot 2023-03-24 at 1 28 47 PM" src="https://user-images.githubusercontent.com/320904/227598542-fcbcb701-010c-4489-a3b7-242612a41374.png">